agent/engine/task_manager.go

```diff
@@ -1141,7 +1142,23 @@ func (mtask *managedTask) startContainerTransitions(transitionFunc containerTran
 func (mtask *managedTask) startResourceTransitions(transitionFunc resourceTransitionFunc) (bool, map[string]string) {
 	anyCanTransition := false
 	transitions := make(map[string]string)
-	for _, res := range mtask.GetResources() {
+
+	resources := mtask.GetResources()
+	anyCanTransition, transitions = mtask.resourceTransitions(resources, transitionFunc, "",
+		anyCanTransition, transitions)
+
+	for _, cont := range mtask.Containers {
+		containerResources := cont.GetResources()
+		anyCanTransition, transitions = mtask.resourceTransitions(containerResources,
+			transitionFunc, "-"+cont.Name, anyCanTransition, transitions)
+	}
```

Comment:
Would the order of calls to resourceTransitions for task and container cause potential issues (feels unlikely but in case there is a container resource transition depends on task resource in certain state).

Is it possible that we merge the container resources and task resources together and call resourceTransitions so that for each loop of resources in `resourceTransitions` we have every resources touched?

Reply:
> Is it possible that we merge the container resources and task resources together and call resourceTransitions so that for each loop of resources in resourceTransitions we have every resource touched?

Either ways the order of resource will be the very similar I think! 

Hmm in what scenario does container resource transition depends on task resource??

Reply:
I take a look again at this and find that transitions are actually happening after this returns, meaning that `anyCanTransition` will contain all resources of which dependencies are resolved, so I guess it should be good.

> In what scenario does container resource transition depends on task resource

I don't have a concrete case for now but since this. My original understanding is that actual status proceed will be happening in this function, in that case if taskResource A depends on containerResource b. and in this case only b can proceed, but since we call resourceTransitions on task first, it will give a response that none of the resources can proceed.

However, since this is only gathering possible transitions and won't return after `resourceTransitions` for task resources, I think this comment is safe to ignore.